tests/test_set.py

```diff
@@ -0,0 +1,58 @@
+import shutil
+
+import numpy as np
+import rasterio
+import xarray as xr
+from affine import Affine
+from numpy.testing import assert_almost_equal
+from rasterio.crs import CRS
+
+from isce2_topsapp.solid_earth_tides import (get_azimuth_time_array,
+                                             update_gunw_with_solid_earth_tide)
+
+
+def test_set_workflow(isce_data_dir_for_set,
+                      gunw_path_for_set,
+                      tmp_path):
+    tmp_gunw = tmp_path / 'temp.nc'
+    shutil.copy(gunw_path_for_set, tmp_gunw)
+
+    update_gunw_with_solid_earth_tide(tmp_gunw, 'reference', isce_data_dir_for_set)
+    # update_gunw_with_solid_earth_tide(tmp_gunw, 'secondary', isce_data_dir_for_set)
+
+    for acq_type in ['reference']:  # , 'secondary']:
+        group = f'/science/grids/corrections/external/tides/solidEarth/{acq_type}'
+        variable = 'solidEarthTide'
+        with rasterio.open(f'netcdf:{tmp_gunw}:{group}/{variable}') as ds:
+            # Check nodata and CRS
+            assert ds.nodata == 0
+            assert ds.crs == CRS.from_epsg(4326)
+
+            # Transform should not be the identity
+            t = ds.transform
+            assert t != Affine(1, 0, 0, 0, 1, 0)
+
+
+def test_azimuth_time(isce_data_dir_for_set, gunw_path_for_set):
+    """Ensures deviation of retrieved azimuth time array is within 1e-3 seconds"""
+    group = 'science/grids/imagingGeometry'
+    with xr.open_dataset(gunw_path_for_set, group=group) as ds:
+        lon_res, lat_res = ds.rio.resolution()
+        lat_res = -lat_res
+
+        # Need upper left corner rather than pixel center
+        hgt = ds.heightsMeta.data
+        lat = ds.latitudeMeta.data + lat_res / 2.
+        lon = ds.longitudeMeta.data - lon_res / 2.
+
+    hgt_mesh, lat_mesh, lon_mesh = np.meshgrid(hgt, lat, lon, indexing='ij')
+    # Azimuth time array
+    X = get_azimuth_time_array(product_dir=isce_data_dir_for_set / 'reference',
+                               height_mesh_arr=hgt_mesh,
+                               latitude_mesh_arr=lat_mesh,
+                               longitude_mesh_arr=lon_mesh)
+    # Total seconds from minimum time
+    Y = (X - X.min()) / np.timedelta64(1, 's')
+    # Vertical standard deviation
+    Y_std = Y.std(axis=0)
+    assert_almost_equal(Y_std, 0, decimal=3)
```

Comment:
@dbekaert - I was able to show that in a vertical column of the azimuth time array that we comute has a standard deviation that never exceeds `1e-3` seconds.

However, if the heights in the `imagingGeometry` group have a max of 9,000 m and min of 1,500 m -- the speed of light takes 3.5e-5 seconds to travel between the highest and lowest level in the cube approximately (or 2 orders of magnitude smaller than the maximum standard deviation in this test).

Reply:
That sounds about right and negligible. We should just make a note of it in the code that this is the assumption we do. If we dont do it thats also ok. There overhead should be small.